### PR TITLE
fix(breadcrumb): disable hasOverflow functionality when ResizeObserver not supported

### LINKED ISSUE #1017: [Breadcrumb] fails to render in browsers without ResizeObserver support
### Describe the bug

The use-resize-observer hook does not check for ResizeObserver support, so is failing to render Breadcrumb stories even when the `hasOverflow` feature is not enabled. Also the `useResizeObserver` story title should mirror the prop name `hasOverflow`.

### To Reproduce

1. In Safari go to https://carbon-addons-iot-react.com/?path=/story/watson-iot-breadcrumb--default

### Expected behavior

This story should render even without ResizeObserver support as `hasOverflow` is false.

### Environment/versions:

- OS: MacOS
- Browser: safari
- `carbon-addons-iot-react` version: 2.57.3

### Additional context

image

### PR DESCRIPTION
**Pull request by tay1orjones**
Closes #1017

**Summary**

- Breadcrumb stories weren't rendering at all in Safari because we don't polyfill ResizeObserver and we also weren't disabling `hasOverflow` in environments with no ResizeObserver support

**Change List (commits, features, bugs, etc)**

- Conditionally assign the ref used for `hasOverflow` based on ResizeObserver support
- In documentation, refactor usage of `useResizeObserver` prop to `hasOverflow`
- Add dev warning for no ResizeObserver support when using `hasOverflow`
- Add test, Update snapshot

**Acceptance Test (how to verify the PR)**

- Breadcrumb stories don't work in Safari here (existing previous build): 
  - https://deploy-preview-1008--carbon-addons-iot-react.netlify.com/?path=/story/watson-iot-breadcrumb--default
- Whereas now this should work in Safari (new build of this PR):
  -  https://deploy-preview-1020--carbon-addons-iot-react.netlify.com/?path=/story/watson-iot-breadcrumb--default

### CODE REVIEW

File: src/components/Breadcrumb/Breadcrumb.jsx [hunk 1]
```diff
@@ -98,7 +109,7 @@ const Breadcrumb = ({ children, className, hasOverflow, ...other }) => {
       className={classnames('breadcrumb--container', {
         'breadcrumb--container__overflowfull': afterOverflowItems.length === 1,
       })}
-      ref={breadcrumbRef}
+      ref={typeof ResizeObserver !== 'undefined' ? breadcrumbRef : null}
```
Comment (bjornalm) [inline src/components/Breadcrumb/Breadcrumb.jsx]: nit: We check this twice now, perhaps put in a function and call it like `browserSupports(ResizeObserver)` to make it  clear that we are dealing with feature detection.

Comment (tay1orjones) [reply src/components/Breadcrumb/Breadcrumb.jsx]: 👍 Great suggestion, done. 

I took a trip down memory lane to see if we could/should use Modernizr but they don't support `esm` and also don't have a feature detection check for `ResizeObserver`. The function is extensible for future detection.

Comment (bjornalm) [reply src/components/Breadcrumb/Breadcrumb.jsx]: Sweet!

**Pull request merged**